module/pollSubmissions/helper.js

```diff
@@ -0,0 +1,335 @@
+                    }
+                
+
+                    let pollDocument = await pollsHelper.pollDocuments
+                    (
+                        {
+                            _id: pollId
+                        },
+                        [
+                            "name"
+                        ]
+                    )
+                
+                    let pollSubmissionDocument = {
+                        pollName: pollDocument[0].name,
+                        pollId: pollId,
+                        submittedAt: new Date(),
+                        responses: responseObject,
+                        isDeleted: false,
+                        userId: userId,
+                        status: messageConstants.common.SUBMISSION_STATUS_COMPLETED
+                    }
+
+                    await database.models.pollSubmissions.create(pollSubmissionDocument);
+                    
+                    let incrementKeyValues = {
+                        numberOfResponses : 1
+                    }
+
+                    Object.values(responseObject).forEach(singleResponse => {
+                        if (Array.isArray(singleResponse.value)) {
+                            singleResponse.value.forEach( value => {
+                                incrementKeyValues[`result.${singleResponse.qid}.${value}`] = 1;
+                            })
+                        }
+                        else {
+                            incrementKeyValues[`result.${singleResponse.qid}.${singleResponse.value}`] = 1;
+                        }
+                    })
+
+                    let updateQuery = { 
+                        $inc: incrementKeyValues
+                    }
+
+                    await pollsHelper.updatePollDocument(pollId, updateQuery)
+            
+                    return resolve({
+                        success: true,
+                        message: messageConstants.apiResponses.POLL_SUBMITTED,
+                        data: true
+                    });
+
+                } else {
+                   throw new Error(messageConstants.apiResponses.POLL_COULD_NOT_BE_SUBMITTED)
+                }
+
+            } catch (error) {
+                return resolve({
+                    success: false,
+                    message: error.message,
+                    data: false
+                });
+            }
+        });
+    }
+
+
+    /**
+    * List all the polls.
+    * @method
+    * @name list
+    * @param {String} userId - userId
+    * @returns {JSON} - Polls list.
+    */
+
+   static list(userId= "") {
+    return new Promise(async (resolve, reject) => {
+        try {
+
+            if(userId == ""){
+                throw new Error(messageConstants.apiResponses.USER_ID_REQUIRED_CHECK);
+            }
+
+            let sortObject = {
+                createdAt: 1
+            }
+
+            let getPollList = [
+                pollsHelper.pollDocuments
```

Comment:
@deepakr123 - Use existing helper function in polls module.

Reply:
resolved